File: ios/ChatAiIos/Models/OpenAiModels.swift
import Foundation

// MARK: - OpenAI API Request & Response Models

public struct OpenAiRequest: Codable {
    public struct Message: Codable {
        public let role: String
        public let content: String
        public init(role: String, content: String) {
            self.role = role
            self.content = content
        }
    }

    public let model: String
    public let messages: [Message]
    public let temperature: Double?

    public init(model: String = "gpt-4o-mini", prompt: String, systemPrompt: String? = nil, temperature: Double? = 0.7) {
        self.model = model
        self.temperature = temperature
        var msgs: [Message] = []
        if let system = systemPrompt, !system.isEmpty {
            msgs.append(Message(role: "system", content: system))
        }
        msgs.append(Message(role: "user", content: prompt))
        self.messages = msgs
    }
}

public struct OpenAiResponse: Codable {
    public struct Choice: Codable {
        public struct Message: Codable {
            public let role: String?
            public let content: String?
        }
        public let message: Message?
        public let finishReason: String?

        enum CodingKeys: String, CodingKey {
            case message
            case finishReason = "finish_reason"
        }
    }

    public let id: String?
    public let choices: [Choice]?

    public var replyText: String? {
        return choices?.first?.message?.content
    }
}

File: app/src/main/java/com/chatai/adr/model/OpenAiModels.kt
package com.chatai.adr.model

import com.google.gson.annotations.SerializedName

data class OpenAiRequest(
    @SerializedName("model")
    val model: String = "gpt-4o-mini",
    @SerializedName("messages")
    val messages: List<Message>,
    @SerializedName("temperature")
    val temperature: Double = 0.7
) {
    data class Message(
        @SerializedName("role")
        val role: String,
        @SerializedName("content")
        val content: String
    )

    companion object {
        fun create(model: String, prompt: String, systemPrompt: String? = null): OpenAiRequest {
            val list = mutableListOf<Message>()
            if (!systemPrompt.isNullOrBlank()) {
                list.add(Message(role = "system", content = systemPrompt))
            }
            list.add(Message(role = "user", content = prompt))
            return OpenAiRequest(model = model, messages = list)
        }
    }
}

data class OpenAiResponse(
    @SerializedName("id")
    val id: String? = null,
    @SerializedName("choices")
    val choices: List<Choice>? = null
) {
    data class Choice(
        @SerializedName("message")
        val message: Message? = null,
        @SerializedName("finish_reason")
        val finishReason: String? = null
    )

    data class Message(
        @SerializedName("role")
        val role: String? = null,
        @SerializedName("content")
        val content: String? = null
    )

    val replyText: String?
        get() = choices?.firstOrNull()?.message?.content
}
